### PR TITLE
Add ExDoc dependency and doc comments for ExFormat module

### PR DESCRIPTION
**Pull request by uohzxela**

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -35,6 +35,6 @@ defmodule ExFormat.Mixfile do
   #
   # Type "mix help deps" for more examples and options
   defp deps do
-    []
+    [{:ex_doc, "~> 0.14", only: :dev, runtime: false}]
```
Comment (whatyouhide) [inline mix.exs]: We're at `~> 0.16` now. Also, no need for `runtime: false`, it's fine if ex_doc makes it to the runtime in the dev environment.

File: lib/ex_format.ex [hunk 2]
```diff
@@ -7,16 +13,60 @@ defmodule ExFormat do
...
+
+  ## Examples
+      iex> ExFormat.format("   quote(do:  foo.bar(1,2,     3))")
+      {:ok, "quote do: foo.bar(1, 2, 3)\\n"}
+
+      iex> ExFormat.format("   quote(do:foo.bar(1,2,     3))")
+      {:error, "nofile:1: keyword argument must be followed by space after: do:"}
+  """
+  @spec format(String.t) :: {:ok, String.t} | {:error, String.t}
+  def format(code_string) do
```
Comment (whatyouhide) [inline lib/ex_format.ex]: No need to have a non-bang function that doesn't raise, if users have the need to "protect" from errors when formatting they can do the rescuing themselves IMO.

Comment (uohzxela) [reply lib/ex_format.ex]: Okay, I'll just revert back to the original code which doesn't do rescue and remove `format!/1`

**Comment by uohzxela** [comment]
@whatyouhide updated

**Review by whatyouhide — APPROVED** [review]

**Pull request merged**